internal/provider/utils/filtering.go

```diff
@@ -97,3 +101,19 @@ func filterByPossibilities[T string | uint16](value T, possibilities []T) (filte
 	}
 	return true
 }
+
+func filterMultipleByPossibilities[T string | uint16](values []T, possibilities []T) (filtered bool) {
```

Comment:
nit since this is only used for []string (categories), no need to have generic for now:

```suggestion
func filterMultipleByPossibilities(values, possibilities []string) (filtered bool) {
```

Reply:
Should I also update `filterByPossibilities` since that's what I copied?